File: 백준/Bronze/25304. 영수증/영수증.js
const fs = require("fs");
const path = require("path");
const input = fs
  .readFileSync(
    process.platform === "linux"
      ? "/dev/stdin"
      : path.join(__dirname, "/input.txt"),
  )
  .toString()
  .trim()
  .split("\n");

const solution = (input) => {
  const X = Number(input[0]);
  const N = Number(input[1]);
  let sum = 0;
  for (let i = 2; i < N + 2; i++) {
    const [a, b] = input[i].split(" ").map(Number);
    sum += a * b;
  }
  console.log(X === sum ? "Yes" : "No");
};

solution(input);

File: 백준/Bronze/25304. 영수증/영수증.cc
#include <iostream>

using namespace std;

int main()
{
    int X;
    cin >> X;
    
    int N;
    cin >> N;

    int sum=0;
    for(int i=0;i<N;i++){
        int a,b;
        cin >> a >> b;
        sum+=(a*b);
    }
    if(X==sum){
        cout << "Yes" << endl;
    }else{
        cout << "No" << endl;
    }

    return 0;
}